### DISCUSSION TITLE
graphql/codegen/docs/guides/angular

### DISCUSSION THREAD

**Discussion by giscus** [dotansimha/graphql-code-generator#9818] [Docs Discussions]
# graphql/codegen/docs/guides/angular

https://the-guild.dev/graphql/codegen/docs/guides/angular

**Comment by J0nnyI** [comment]
a git repository would be a great help, since I have seemingly no way to find what causes 'PostsGQL' to be generated or where it comes from

**Reply by avgustine** [reply]
Yeah, really difficult to figure out what suppose to be in generated file.
Also having this issue
Unable to find any GraphQL type definitions for the following pointers:
[FAILED]
[FAILED]           - ./src/**/*.ts

**Reply by arron21** [reply]
The graphql-codegen requires that you have graphQL queries already written out. The CodeGenConfig is going to look for your graphQL queries in this filepath match `documents: './src/**/*.ts',`

For example the following query is in my `/src/app/graphql/region.graph.ts file` 

```
import { gql } from 'apollo-angular';

export const GET_REGIONS = gql`
  query GetRegions {
    regions {
      id
      name
    }
  }
`;
```

Then when I run `graphql-codegen` it will take that information and generate a new file at `./src/generated/newGraphQLCode.ts`
```
 generates: {
    './src/generated/newGraphQLCode.ts': {
      plugins: [
        'typescript',
        'typescript-operations',
        'typescript-apollo-angular',
      ],
    },
  },
```

The code it generates looks like this

```
export const GetRegionsDocument = gql`
  query GetRegions {
    regions {
      id
      name
    }
  }
`;

@Injectable({
  providedIn: 'root',
})
export class GetRegionsGQL extends Apollo.Query<
  GetRegionsQuery,
  GetRegionsQueryVariables
> {
  override document = GetRegionsDocument;

  constructor(apollo: Apollo.Apollo) {
    super(apollo);
  }
}
``` 

I can then change my component code to this
```
export class RegionsComponent {
  response: any;
  constructor(getRegionsGQL: GetRegionsGQL) {
    this.response = getRegionsGQL
    .watch()
    .valueChanges.pipe(map((result) => result.data.regions));
  }
}
```

and the html
```
<app-list link="region" [items]="response | async"></app-list>
```

**Comment by ext-vuillaume** [comment]
Is there any reported issue concerning duplicates in generated.ts  ?
I have a query that is generated twice even tho I don't have any duplicated queries either in my schema.graphqls nor in my gql queries...
